### PR TITLE
docs: Update changelog for 3.1 release

### PR DESCRIPTION
**Pull request by cloudxxx8**
**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Comment by cloudxxx8** [comment]
@lenny-intel I don't have the permission to assign the reviewer.  Could you please help review this PR?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+
+## [v3.1.0] Napa - 2023-11-15 (Only compatible with the 3.x releases)
+
+### ✨  Features
+
+- Add command line/environment flag for commonConfig ([#486](https://github.com/edgexfoundry/device-sdk-c/issues/486)) ([753c0e6…](https://github.com/edgexfoundry/device-sdk-c/commit/753c0e66273d645c82de476ca0a438ee98891f55))
+- Load common configuration from config provider ([#484](https://github.com/edgexfoundry/device-sdk-c/issues/484)) ([4c52de3…](https://github.com/edgexfoundry/device-sdk-c/commit/4c52de37db7d3bd0277fba2739b75d0a1bd475b3))
+```text
+
+BREAKING CHANGE: load common configuration from config provider ([#484](https://github.com/edgexfoundry/device-sdk-c/issues/484))
```
Comment (lenny-intel) [inline CHANGELOG.md:13]: This shouldn't be a breaking change. One could still have all the configuration in the local private config.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (cloudxxx8) [reply CHANGELOG.md:13]: the default behavior is not load the all configuration from local.
it's the problem we didn't accomplish this feature C v3.0
I am removing the breaking change anyway.

**Comment by cloudxxx8** [comment]
recheck

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**